skrf/media/cpw.py

```diff
@@ -146,20 +147,39 @@ def alpha_conductor(self) -> NumberLike:
         --------
         surface_resistivity : calculates surface resistivity
         """
-        if self.rho is None or self.t is None:
-            raise(AttributeError('must provide values conductivity and conductor thickness to calculate this. see initializer help'))
-
-        t, k1, ep_re = self.t, self.k1,self.ep_re
-        r_s = surface_resistivity(f=self.frequency.f, rho=self.rho, \
-                mu_r=1)
+        t, k1, ep_re, f = self.t, self.k1,self.ep_re, self.frequency.f
+        rho = self.rho
+        
+        if t is not None and t > 0:
+            if rho is None:
+                raise(AttributeError('must provide values conductivity and conductor thickness to calculate this. see initializer help'))
+            
+            r_s = surface_resistivity(f=f, rho=rho, \
+                    mu_r=1)
+            ds = skin_depth(f = f, rho = rho, mu_r = 1.)
+            if(any(t < 3 * ds)):
+                warnings.warn(
+                    'Conductor loss calculation invalid for line'
+                    'height t ({})  < 3 * skin depth ({})'.format(t, ds[0]),
+                    RuntimeWarning
+                    )
+                
+            a = self.w/2.
+            b = self.s+self.w/2.
+            K = ellipk      # complete elliptical integral of first kind
+            K_p = lambda x: ellipk(sqrt(1-x**2)) # ellipk's compliment
+            a_c = ((r_s * sqrt(ep_re)/(480*pi*K(k1)*K_p(k1)*(1-k1**2) ))*\
+                    (1./a * (pi+log((8*pi*a*(1-k1))/(t*(1+k1)))) +\
+                     1./b * (pi+log((8*pi*b*(1-k1))/(t*(1+k1))))))
+        else:
+            a_c = zeros(f.shape)
+            
         a = self.w/2.
         b = self.s+self.w/2.
         K = ellipk      # complete elliptical integral of first kind
         K_p = lambda x: ellipk(sqrt(1-x**2)) # ellipk's compliment
```

Comment:
These four lines are not necessary any more

Reply:
Indeed, let's get rid of them !